internal/provider/resource_virtual_machine_test.go

```diff
@@ -583,6 +592,105 @@ func TestAccKatapultVirtualMachine_update_ips(t *testing.T) {
 	})
 }
 
+func TestAccKatapultVirtualMachine_update_group(t *testing.T) {
+	tt := newTestTools(t)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeAggregateTestCheckFunc(
+			testAccCheckKatapultVirtualMachineDestroy(tt),
+			testAccCheckKatapultIPDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: undent.String(`
+					resource "katapult_ip" "web" {}
+
+					resource "katapult_virtual_machine_group" "web" {
+						name = "vm-group"
+					}
+
+					resource "katapult_virtual_machine" "base" {
+						package       = "rock-3"
+						disk_template = "ubuntu-18-04"
+						disk_template_options = {
+							install_agent = true
+						}
+						ip_address_ids = [
+							katapult_ip.web.id,
+						]
+					}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckNoResourceAttr(
+						"katapult_virtual_machine.base",
+						"group_id",
+					),
+				),
+			},
+			{
+				Config: undent.String(`
+					resource "katapult_ip" "web" {}
+
+					resource "katapult_virtual_machine_group" "web" {
+						name = "vm-group"
+					}
+
+					resource "katapult_virtual_machine" "base" {
+						package       = "rock-3"
+						disk_template = "ubuntu-18-04"
+						disk_template_options = {
+							install_agent = true
+						}
+						ip_address_ids = [
+							katapult_ip.web.id,
+						]
+						group_id = katapult_virtual_machine_group.web.id
+					}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckResourceAttrPair(
+						"katapult_virtual_machine.base", "group_id",
+						"katapult_virtual_machine_group.web", "id",
+					),
+				),
+			},
+			{
+				Config: undent.String(`
+				resource "katapult_ip" "web" {}
+
+				resource "katapult_virtual_machine" "base" {
+					package       = "rock-3"
+					disk_template = "ubuntu-18-04"
+					disk_template_options = {
+						install_agent = true
+					}
+					ip_address_ids = [
+						katapult_ip.web.id,
+					]
+				}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_virtual_machine.base",
+						"group_id", "",
+					),
```

Comment:
I'd prefer to use `resource.TestCheckNoResourceAttr` here but unfortunately it gets set to an empty string 🤔

Reply:
Hmm, I'm just guessing off the top of my head, but I think that might be fixable in the Read function when setting `group_id`:

```go
	if vm.Group != nil && vm.Group != katapult.NullVirtualMachineGroup {
 		_ = d.Set("group_id", vm.Group.ID)
 	}
```

Reply:
I'm not sure if `TestCheckNoResourceAttr` checks for the key, and I cannot find any way to delete a key. `Set` will only update the existing state, I think.

Reply:
Yeah, reading the docs it's about ensuring no value exists. I think it's fine to just ensure the value is a empty string, as we, and Go, treats that as not set.